backend/src/main/java/de/bund/digitalservice/ris/norms/application/service/NormService.java

```diff
@@ -438,15 +563,30 @@ private Zielnorm createZielNormen(final Zielnorm zielnorm) {
         )
         .toList()
     );
-    // 1. New manifestation for becoming gegenstandslos --> isCreated = true && isGegenstandslos = true && createdBy = OTHER_VERKUENDUNG
-
-    // 2. New expressions replacing those set to gegenstandslos --> isCreated = false && isGegenstandslos = false && createdBy = SYSTEM
-
-    // 3. Completely new expressions --> isCreated = false && isGegenstandslos = false && createdBy = THIS_VERKUENDUNG
-
-    // 4. Add elis of new expressions from 2. and 3. into XML node amended-expressions
-
-    // 5. Orphan elements in amended-expressions? meaning present there but not in passed "zielnormen"? Remove XML from DB
+  }
 
+  /**
+   * Get the previous closest non-gegenstandslos norm to the passed date.
+   */
+  private Optional<Norm> findPreviousClosestExistingExpression(
+    NormWorkEli zielnormWorkEli,
+    LocalDate dateForNewExpression
+  ) {
+    return loadNormExpressionElisPort
+      .loadNormExpressionElis(new LoadNormExpressionElisPort.Options(zielnormWorkEli))
+      .stream()
+      .filter(eli -> eli.getPointInTime().isBefore(dateForNewExpression))
+      .map(eli ->
+        new AbstractMap.SimpleEntry<>(eli, loadNormPort.loadNorm(new LoadNormPort.Options(eli)))
+      )
+      .filter(entry -> entry.getValue().isPresent()) // keep only existing Norms
+      .map(entry -> new AbstractMap.SimpleEntry<>(entry.getKey(), entry.getValue().get()))
+      .filter(entry -> !entry.getValue().isGegenstandlos())
+      .max(
+        Comparator.comparing((Map.Entry<NormExpressionEli, Norm> entry) ->
+          entry.getKey().getPointInTime()
+        ).thenComparing(entry -> entry.getKey().getVersion())
```

Comment:
i think the elis are implementing Comparable and do exactly this sorting by default